Source/SwiftLintBuiltInRules/Rules/Style/TrailingClosureRule.swift

```diff
@@ -18,127 +19,78 @@ struct TrailingClosureRule: OptInRule {
+                violations.append(node.positionAfterSkippingLeadingTrivia)
+            } else {
+                guard node.lastArgumentIsClosure, node.isNotNextToLeftBrace else { return }
 
-        if let kind = dictionary.statementKind, kind != .brace {
-            // trailing closures are not allowed in `if`, `guard`, etc
-            results += dictionary.substructure.flatMap { subDict -> [ByteCount] in
-                guard subDict.statementKind == .brace else {
-                    return []
+                if node.shouldTrigger {
+                    violations.append(node.positionAfterSkippingLeadingTrivia)
                 }
-
-                return violationOffsets(for: subDict, file: file)
-            }
-        } else {
-            results += dictionary.substructure.flatMap { subDict in
-                violationOffsets(for: subDict, file: file)
             }
         }
 
-        return results
-    }
-
-    private func shouldBeTrailingClosure(dictionary: SourceKittenDictionary, file: SwiftLintFile) -> Bool {
-        func shouldTrigger() -> Bool {
-            return !isAlreadyTrailingClosure(dictionary: dictionary, file: file) &&
-                !isAnonymousClosureCall(dictionary: dictionary, file: file)
-        }
-
-        let arguments = dictionary.enclosedArguments
-
-        // check if last parameter should be trailing closure
-        if !configuration.onlySingleMutedParameter, arguments.isNotEmpty,
-            case let closureArguments = filterClosureArguments(arguments, file: file),
-            closureArguments.count == 1,
-            closureArguments.last?.offset == arguments.last?.offset {
-            return shouldTrigger()
+        override func visit(_ node: ConditionElementListSyntax) -> SyntaxVisitorContinueKind {
+            .skipChildren
         }
 
-        let argumentsCountIsExpected: Bool = {
-            if SwiftVersion.current >= .fiveDotSix, arguments.count == 1,
-               arguments[0].expressionKind == .argument {
-                return true
-            }
-
-            return arguments.isEmpty
-        }()
-        // check if there's only one unnamed parameter that is a closure
-        if argumentsCountIsExpected,
-            let offset = dictionary.offset,
-            let totalLength = dictionary.length,
-            let nameOffset = dictionary.nameOffset,
-            let nameLength = dictionary.nameLength,
-            case let start = nameOffset + nameLength,
-            case let length = totalLength + offset - start,
-            case let byteRange = ByteRange(location: start, length: length),
-            let range = file.stringView.byteRangeToNSRange(byteRange),
-            let match = regex("\\s*\\(\\s*\\{").firstMatch(in: file.contents, options: [], range: range)?.range,
-            match.location == range.location {
-            return shouldTrigger()
+        override func visit(_ node: ForStmtSyntax) -> SyntaxVisitorContinueKind {
+            .skipChildren
```

Comment:
I don't think the following example would trigger with this change:

```swift
for n in list {
    ↓n.forEach({ print($0) })
}
```

Think about how to keep traversing the loop's body while skipping everything else. 😉

Reply:
Fixed at 4636f40

(Finally understanding the point of the `walk()` function. It is pretty useful. 😀)